src/Control/Distributed/Spark/RDD.hs

```diff
@@ -79,6 +91,25 @@ map clos rdd = do
     f <- reflect clos
     call rdd "map" [coerce f]
 
+mapPartitions
+  :: (Reflect (Closure (Int32 -> Stream (Of a) IO () -> Stream (Of b) IO ())) ty, Typeable a, Typeable b)
+  => Choice "preservePartitions"
+  -> Closure (Stream (Of a) IO () -> Stream (Of b) IO ())
+  -> RDD a
+  -> IO (RDD b)
+mapPartitions preservePartitions clos rdd =
+  mapPartitionsWithIndex preservePartitions (closure (static const) `cap` clos) rdd
```

Comment:
I couldn't get simpler constraints to work. If you have an idea how to that would be awesome. I couldn't quite figure out why the simpler constraints didn't work. Looked to me like it came down to incompleteness of GHC's constraint solver.